pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2466,32 +2462,64 @@ private String receiveCommandStatus() throws IOException {
     return status;
   }
 
+  private static boolean isDigitAt(String status, int i) {
+    return i > 0 && i < status.length() && Character.isDigit(status.charAt(i));
+  }
+
+  private static int digitAt(String s, int pos) {
+    int c = s.charAt(pos) - '0';
+    if (c < 0 || c > 9) {
+      throw new NumberFormatException("Input string: \"" + s + "\"");
+    }
+    return c;
+  }
+
+  private static long toLong(String s, int beginIndex, int endIndex) {
+    if (endIndex > 8 && !isDigitAt(s, beginIndex)) {
+      return Long.parseLong(s.substring(beginIndex, beginIndex));
+    }
+    long res = digitAt(s, beginIndex);
+    for (beginIndex++; beginIndex < endIndex; beginIndex++) {
+      res = res * 10 + digitAt(s, beginIndex);
+    }
+    return res;
+  }
+
   private void interpretCommandStatus(String status, ResultHandler handler) {
     long oid = 0;
     long count = 0;
-    Matcher matcher = COMMAND_COMPLETE_PATTERN.matcher(status);
-    if (matcher.matches()) {
-      // String command = matcher.group(1);
-      String group2 = matcher.group(2);
-      String group3 = matcher.group(3);
+
+    // This code processes the CommandComplete (B) message.
+    // Status is in the format of "COMMAND OID ROWS" where both 'OID' and 'ROWS' are optional
+    // and COMMAND can have spaces within it, like CREATE TABLE.
+    // Scan backwards, while searching for a maximum of two number groups
+    // COMMAND OID ROWS
+    // COMMAND ROWS
+    // Assumption: command does not contain digits
```

Comment:
Is it a valid assumption that no command will have a number (like end in 2 for a second/different behavior)?